File: sandbox/exercism/ruby/2024/space_age.rb
# https://exercism.org/tracks/ruby/exercises/space-age

class SpaceAge
  OrbitalPeriod = Data.define(:planet, :period)

  SECONDS_PER_YEAR_ON_EARTH = 31557600
  ORBITAL_PERIODS = [
    OrbitalPeriod.new(:earth,   SECONDS_PER_YEAR_ON_EARTH),
    OrbitalPeriod.new(:mercury, SECONDS_PER_YEAR_ON_EARTH *   0.2408467),
    OrbitalPeriod.new(:venus,   SECONDS_PER_YEAR_ON_EARTH *   0.61519726),
    OrbitalPeriod.new(:mars,    SECONDS_PER_YEAR_ON_EARTH *   1.8808158),
    OrbitalPeriod.new(:jupiter, SECONDS_PER_YEAR_ON_EARTH *  11.862615),
    OrbitalPeriod.new(:saturn,  SECONDS_PER_YEAR_ON_EARTH *  29.447498),
    OrbitalPeriod.new(:uranus,  SECONDS_PER_YEAR_ON_EARTH *  84.016846),
    OrbitalPeriod.new(:neptune, SECONDS_PER_YEAR_ON_EARTH * 164.79132),
  ].freeze

  def initialize(seconds)
    @seconds = seconds.to_f
  end

  ORBITAL_PERIODS.each do |op|
    define_method("on_#{op.planet}") { @seconds / op.period }
  end
end

File: sandbox/exercism/go/2025/space_age.go
// https://exercism.org/tracks/go/exercises/space-age

package space

type Planet string

const (
	Mercury Planet = "Mercury"
	Venus   Planet = "Venus"
	Earth   Planet = "Earth"
	Mars    Planet = "Mars"
	Jupiter Planet = "Jupiter"
	Saturn  Planet = "Saturn"
	Uranus  Planet = "Uranus"
	Neptune Planet = "Neptune"
)

const SecondsPerYearOnEarth = 31557600

func Age(seconds float64, planet Planet) float64 {
	ages := map[Planet]float64{
		Mercury: SecondsPerYearOnEarth * 0.2408467,
		Venus:   SecondsPerYearOnEarth * 0.61519726,
		Earth:   SecondsPerYearOnEarth,
		Mars:    SecondsPerYearOnEarth * 1.8808158,
		Jupiter: SecondsPerYearOnEarth * 11.862615,
		Saturn:  SecondsPerYearOnEarth * 29.447498,
		Uranus:  SecondsPerYearOnEarth * 84.016846,
		Neptune: SecondsPerYearOnEarth * 164.79132,
	}

	age, ok := ages[planet]

	if !ok { return -1.000000 }

	return seconds / age
}
